wcomponents-core/src/main/java/com/github/bordertech/wcomponents/WMultiFileWidget.java

```diff
@@ -670,6 +691,25 @@ protected void doHandleUploadRequest(final Request request) {
 
 		// Wrap the file item
 		FileItemWrap wrap = new FileItemWrap(items[0]);
+		
+		// if fileType is supplied then validate it
+		if (isSetFileTypes()) {
```

Comment:
You could combine these two if statements (I have been pinged by static analysis on this before)